packages/blitz-auth/src/client/index.tsx

```diff
@@ -347,8 +340,12 @@ function withBlitzAuthPlugin<TProps = any>(Page: ComponentType<TProps> | BlitzPa
           authenticate &&
           typeof authenticate === "object" &&
           authenticate.redirectTo &&
-          authenticate.role &&
-          !authorizeRole(authenticate.role, publicData.role as string)
+          authenticate.roles &&
+          (!Array.isArray(authenticate.roles)
+            ? authorizeRole(authenticate.roles, publicData.roles as string[])
+            : authenticate.roles.some((role: string) =>
+                authorizeRole(role, publicData.roles as string[]),
+              ))
```

Comment:
Here a possible implementation, since `authorizeRole` can handle an array of roles
```suggestion
          authenticate.role &&
          !authorizeRole(authenticate.role, [
          publicData.role,
          ...(Array.isArray(publicData.roles) ? publicData.roles : []),
        ] as string[])
```